### ISSUE TITLE
How to train Image Manipulation Trace Feature Extractor through Fine-Grained Manipulation（IMC-385）?

### ISSUE THREAD

**Issue by Zhengtq** [RonyAbecidan/ManTraNet-pytorch#5]
I find that there is no downsampling in Image Manipulation Trace Feature Extractor. In other words, the output of Image Manipulation Trace Feature Extractor is the sample size as the input image.
So How to train Image Manipulation Trace Feature Extractor through Fine-Grained Manipulation(IMC-385) as claimed in paper?

**Comment by RonyAbecidan** [comment]
Hello, in practice, to train the IMTF like the authors you should construct a database with 385 different forgeries that you will ask to the IMTF to predict. Even if it is not written explicitly in the paper, I think the authors have just plugged to the IMTF a dense layer with an output dimension of 385 followed by a softmax to train it. Then, they kept only the weights of the IMTF.

To make things easier for you, I am going soon to decouple the IMTF from the Anomaly detector so that they can be trained separately as the authors did.

**Comment by RonyAbecidan** [comment]
This comment was for the MantraNet architecture. If you see the scheme of the architecture you will see that indeed, there is no dense layers in Mantranet.

However, to train the IMC-385 at first place, they added a dense layer to find the weights. See the decision block in their training architectures :

![](https://cdn.mathpix.com/snip/images/FCSAha-FtxSuZ24xU5lQXNNCbIknC-L4HSKGYk9CXJo.original.fullsize.png)

**Comment by RonyAbecidan** [comment]
Re-reading the paper, I think you're right ! (See the decision block of their IMC-385 below)

![](https://cdn.mathpix.com/snip/images/FCSAha-FtxSuZ24xU5lQXNNCbIknC-L4HSKGYk9CXJo.original.fullsize.png)

**Comment by Zhengtq** [comment]
Very thanks for your replay!
